app/scripts/controllers/permissions/caveat-mutators.js

```diff
@@ -24,7 +25,8 @@ export const CaveatMutatorFactories = {
  */
 function removeAccount(targetAccount, existingAccounts) {
   const newAccounts = existingAccounts.filter(
-    (address) => address !== targetAccount,
+    (address) =>
+      toChecksumAddress(address) !== toChecksumAddress(targetAccount),
```

Comment:
Could we compute the checksum of the `targetAccount` outside the loop?

Reply:
Thanks, thats a good catch!